internal/listener/listener.go

```diff
@@ -34,13 +34,26 @@ func NewListener(db *database.DB, runtimeConfig *config.RuntimeConfig, logs *log
 		logs:          logs,
 		runtimeConfig: runtimeConfig,
 	}
+
+	debugMux := http.NewServeMux()
+	debugMux.HandleFunc("/dump-config", l.DumpConfig)
+	debugMux.HandleFunc("/dump-incidents", l.DumpIncidents)
+	debugMux.HandleFunc("/dump-schedules", l.DumpSchedules)
+
+	l.mux.Handle("/debug/", http.StripPrefix("/debug", l.debugAuth(debugMux)))
 	l.mux.HandleFunc("/process-event", l.ProcessEvent)
-	l.mux.HandleFunc("/dump-config", l.DumpConfig)
-	l.mux.HandleFunc("/dump-incidents", l.DumpIncidents)
-	l.mux.HandleFunc("/dump-schedules", l.DumpSchedules)
 	return l
 }
 
+func (l *Listener) debugAuth(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !l.checkDebugPassword(w, r) {
```

Comment:
With introducing this wrapper, the check can be removed from the handler methods `DumpConfig`, `DumpIncidents` and `DumpSchedules`.

Afterwards, this would be the only position where `checkDebugPassword` would be called. If you want to, you could inline this method. But this is not necessary. Do as you please.